### PR TITLE
DAOS-16251 tests: Fix various buffer overflows

### PR DESCRIPTION
**Pull request by liw**
- vos: Fix vtx_pm buffer overflow
  - vos: Fix evt_ctl stack buffer overflow
  - object: Fix srv_checksum_tests buffer overflow
  - utils: Fix ddb_vos_tests buffer overflow etc

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/object/tests/srv_checksum_tests.c [hunk 1]
```diff
@@ -1275,7 +1275,7 @@ fetch_with_hole7(void **state)
 		.layout = {
 			{.data = "ABCDEFGHIJKLM", .sel = {1, 2}, .ful = {1, 13} },
 			{.data = "", .sel = {3, 7}, .ful = {3, 7}, .is_hole = true},
-			{.data = "HIJKLM", .sel = {8, 13}, .ful = {1, 13} },
+			{.data = "HIJKLM", .sel = {8, 13}, .ful = {8, 13} },
```
Comment (knard-intel) [inline src/object/tests/srv_checksum_tests.c:1278]: Not sure to understand this one: not enough background.

**Review by knard38 — APPROVED** [review]
LGTM for what I understand.

File: src/object/tests/srv_checksum_tests.c (hunk 1 above)
Comment (ryon-jensen) [reply src/object/tests/srv_checksum_tests.c:1278]: These are defining ranges for extents ... selected vs full extent. Change looks good.

**Review by ryon-jensen — APPROVED** [review]
lgtm

**Pull request merged**